libpod/container_api.go

```diff
@@ -771,6 +771,12 @@ func (c *Container) WaitForConditionWithInterval(ctx context.Context, waitTimeou
 						// This allows callers to actually wait for the ctr to be removed.
 						if wantedStates[define.ContainerStateRemoving] &&
 							(errors.Is(err, define.ErrNoSuchCtr) || errors.Is(err, define.ErrCtrRemoved)) {
+							// check if the exit code was recorded in the db to return it
+							exitCode, err := c.runtime.state.GetContainerExitCode(c.ID())
+							if err == nil {
+								trySend(exitCode, nil)
```

Comment:
Drive-by-comment:  do we need a `return` after the trySend() here?

Reply:
It likely should yes, however looking at the code it is not a bug. The channel below will always read the first result and then exit. If there is a second result written below it will get simply ignored.